#include "commande_widget.h"
#include "ui_commande_widget.h"
#include <QApplication>
#include <QMessageBox>
#include <QTableWidgetItem>
#include <QHeaderView>
#include <QTextCharFormat>
#include <QColor>
#include <QDialog>
#include <QVBoxLayout>
#include <QHBoxLayout>
#include <QLabel>
#include <QLineEdit>
#include <QTextEdit>
#include <QPushButton>
#include <QFormLayout>
#include <QComboBox>
#include <QDateEdit>
#include <QDoubleSpinBox>
#include <QCheckBox>
#include <QGroupBox>
#include <QTabWidget>
#include <QTableWidget>
#include <QCalendarWidget>

CommandeWidget::CommandeWidget(QWidget *parent)
    : QWidget(parent)
    , ui(new Ui::CommandeWidget)
{
    ui->setupUi(this);

    // Cr√©er les actions programmatiquement
    QAction *actionNew = new QAction("&Nouvelle Commande", this);
    actionNew->setObjectName("actionNew");
    actionNew->setShortcut(Qt::CTRL | Qt::Key_N);

    QAction *actionEdit = new QAction("&Modifier", this);
    actionEdit->setObjectName("actionEdit");
    actionEdit->setShortcut(Qt::CTRL | Qt::Key_E);

    QAction *actionDelete = new QAction("&Supprimer", this);
    actionDelete->setObjectName("actionDelete");
    actionDelete->setShortcut(Qt::Key_Delete);

    QAction *actionEnvoyerEmail = new QAction("&Envoyer Email", this);
    actionEnvoyerEmail->setObjectName("actionEnvoyerEmail");
    actionEnvoyerEmail->setShortcut(Qt::CTRL | Qt::SHIFT | Qt::Key_E);

    QAction *actionExportPDF = new QAction("&Exporter PDF", this);
    actionExportPDF->setObjectName("actionExportPDF");
    actionExportPDF->setShortcut(Qt::CTRL | Qt::Key_P);

    QAction *actionRefresh = new QAction("&Actualiser", this);
    actionRefresh->setObjectName("actionRefresh");
    actionRefresh->setShortcut(Qt::Key_F5);

    QAction *actionStatistics = new QAction("üìä Statistiques", this);
    actionStatistics->setObjectName("actionStatistics");
    actionStatistics->setShortcut(Qt::CTRL | Qt::Key_G);

    QAction *actionCalendar = new QAction("üìÖ Calendrier", this);
    actionCalendar->setObjectName("actionCalendar");
    actionCalendar->setShortcut(Qt::CTRL | Qt::Key_L);

    // Ajouter les actions √† la toolbar (Statistiques, Calendrier, Export et Refresh)
    if (ui->toolBar) {
        ui->toolBar->addAction(actionStatistics);
        ui->toolBar->addAction(actionCalendar);
        ui->toolBar->addSeparator();
        ui->toolBar->addAction(actionExportPDF);
        ui->toolBar->addAction(actionRefresh);
    }

    // Stocker les pointeurs comme propri√©t√©s pour acc√®s dans setupConnections
    setProperty("actionNew", QVariant::fromValue((QObject*)actionNew));
    setProperty("actionEdit", QVariant::fromValue((QObject*)actionEdit));
    setProperty("actionDelete", QVariant::fromValue((QObject*)actionDelete));
    setProperty("actionEnvoyerEmail", QVariant::fromValue((QObject*)actionEnvoyerEmail));
    setProperty("actionExportPDF", QVariant::fromValue((QObject*)actionExportPDF));
    setProperty("actionRefresh", QVariant::fromValue((QObject*)actionRefresh));
    setProperty("actionStatistics", QVariant::fromValue((QObject*)actionStatistics));
    setProperty("actionCalendar", QVariant::fromValue((QObject*)actionCalendar));

    // Connecter les actions aux slots
    connect(actionNew, &QAction::triggered, this, &CommandeWidget::onActionNew);
    connect(actionEdit, &QAction::triggered, this, &CommandeWidget::onActionEdit);
    connect(actionDelete, &QAction::triggered, this, &CommandeWidget::onActionDelete);
    connect(actionEnvoyerEmail, &QAction::triggered, this, &CommandeWidget::onActionSendEmail);
    connect(actionExportPDF, &QAction::triggered, this, &CommandeWidget::onActionExportPDF);
    connect(actionRefresh, &QAction::triggered, this, &CommandeWidget::onActionRefresh);
    connect(actionStatistics, &QAction::triggered, this, &CommandeWidget::onActionStatistics);
    connect(actionCalendar, &QAction::triggered, this, &CommandeWidget::onActionCalendar);

    // ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
    // Configuration des connexions et donn√©es
    // ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
    setupConnections();
    setupTableData();
    updateStatistics();

    // Configurer le calendrier
    QDate today = QDate::currentDate();
    ui->dateFrom->setDate(today.addDays(-30));
    ui->dateTo->setDate(today);
}

CommandeWidget::~CommandeWidget()
{
    delete ui;
}

void CommandeWidget::setupConnections()
{
    // Les actions ont d√©j√† √©t√© connect√©es dans le constructeur
    
    // Connecter les boutons d'actions rapides
    connect(ui->btnNewCommande, &QPushButton::clicked, this, &CommandeWidget::onBtnNewCommandeClicked);
    connect(ui->btnEditCommande, &QPushButton::clicked, this, &CommandeWidget::onBtnEditCommandeClicked);
    connect(ui->btnDeleteCommande, &QPushButton::clicked, this, &CommandeWidget::onBtnDeleteCommandeClicked);
    connect(ui->btnEnvoyerEmail, &QPushButton::clicked, this, &CommandeWidget::onBtnEnvoyerEmailClicked);

    // Connecter les fonctionnalit√©s avanc√©es
    // (Les groupBox de statistiques ont √©t√© supprim√©s - affich√©s maintenant dans la fen√™tre Statistiques)

    // Connecter la s√©lection dans la table
    connect(ui->tableCommandes, &QTableWidget::itemSelectionChanged, this, &CommandeWidget::updateQuickInfo);

    // Connecter les filtres
    connect(ui->lineEditSearch, &QLineEdit::textChanged, this, &CommandeWidget::onSearchTextChanged);
    connect(ui->comboFournisseur, &QComboBox::currentTextChanged, this, &CommandeWidget::onFilterChanged);
    connect(ui->comboStatus, &QComboBox::currentTextChanged, this, &CommandeWidget::onFilterChanged);
    connect(ui->dateFrom, &QDateEdit::dateChanged, this, &CommandeWidget::onFilterChanged);
    connect(ui->dateTo, &QDateEdit::dateChanged, this, &CommandeWidget::onFilterChanged);
}

void CommandeWidget::setupTableData()
{
    // Configurer la table avec des donn√©es de test
    QStringList headers = {"ID", "Libell√©", "Fournisseur", "√âtat", "Date Arriv√©e", "Date Sortie", "Montant Total"};
    ui->tableCommandes->setColumnCount(headers.size());

    for (int i = 0; i < headers.size(); ++i) {
        QTableWidgetItem *header = new QTableWidgetItem(headers[i]);
        ui->tableCommandes->setHorizontalHeaderItem(i, header);
    }

    // Donn√©es de test
    QStringList commandesData = {
        "CMD001,Huile d'olive vierge extra,Fournisseur A,En cours,2024-01-15,2024-01-20,1500.00‚Ç¨",
        "CMD002,Presse hydraulique 20T,Fournisseur B,Valid√©e,2024-01-10,2024-01-25,2300.00‚Ç¨",
        "CMD003,Filtres √† huile premium,Fournisseur C,Pr√™te,2024-01-05,2024-01-30,1800.00‚Ç¨",
        "CMD004,Bouteilles verre 500ml,Fournisseur A,Annul√©e,2024-01-12,2024-01-12,900.00‚Ç¨",
        "CMD005,Machine d'embouteillage,Fournisseur D,En cours,2024-01-18,,3500.00‚Ç¨",
        "CMD006,√âtiquettes personnalis√©es,Fournisseur B,Pr√™te,2024-01-08,2024-01-28,650.00‚Ç¨"
    };

    ui->tableCommandes->setRowCount(commandesData.size());

    for (int row = 0; row < commandesData.size(); ++row) {
        QStringList fields = commandesData[row].split(',');
        for (int col = 0; col < fields.size() && col < headers.size(); ++col) {
            QTableWidgetItem *item = new QTableWidgetItem(fields[col]);
            ui->tableCommandes->setItem(row, col, item);
        }
    }

    // Ajuster la largeur des colonnes
    ui->tableCommandes->resizeColumnsToContents();
}

// ==================== CRUD ACTIONS ====================

void CommandeWidget::onActionNew()
{
    showCrudDialog("Nouvelle Commande");
}

void CommandeWidget::onActionEdit()
{
    if (ui->tableCommandes->currentRow() >= 0) {
        showCrudDialog("Modifier Commande");
    } else {
        QMessageBox::warning(this, "Aucune s√©lection",
                             "Veuillez s√©lectionner une commande √† modifier.");
    }
}

void CommandeWidget::onActionDelete()
{
    int currentRow = ui->tableCommandes->currentRow();
    if (currentRow >= 0) {
        QMessageBox::StandardButton reply;
        reply = QMessageBox::question(this, "Confirmation",
                                      "√ätes-vous s√ªr de vouloir supprimer cette commande ?",
                                      QMessageBox::Yes | QMessageBox::No);

        if (reply == QMessageBox::Yes) {
            ui->tableCommandes->removeRow(currentRow);
            updateStatistics();
            QMessageBox::information(this, "Succ√®s", "Commande supprim√©e avec succ√®s.");
        }
    } else {
        QMessageBox::warning(this, "Aucune s√©lection",
                             "Veuillez s√©lectionner une commande √† supprimer.");
    }
}

void CommandeWidget::onActionSendEmail()
{
    QDialog *emailDialog = new QDialog(this);
    emailDialog->setWindowTitle("Envoyer Email");
    emailDialog->setMinimumSize(500, 400);

    QVBoxLayout *layout = new QVBoxLayout(emailDialog);

    QLabel *label = new QLabel("Interface d'envoi d'email", emailDialog);
    QTextEdit *messageEdit = new QTextEdit(emailDialog);
    messageEdit->setPlaceholderText("√âcrivez votre message ici...");
    messageEdit->setMinimumHeight(200);

    QPushButton *sendBtn = new QPushButton("Envoyer", emailDialog);
    QPushButton *cancelBtn = new QPushButton("Annuler", emailDialog);

    QHBoxLayout *buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(sendBtn);
    buttonLayout->addWidget(cancelBtn);

    layout->addWidget(label);
    layout->addWidget(messageEdit);
    layout->addLayout(buttonLayout);

    connect(sendBtn, &QPushButton::clicked, emailDialog, [emailDialog, messageEdit]() {
        QMessageBox::information(emailDialog, "Email envoy√©",
                                 "L'email a √©t√© envoy√© avec succ√®s.");
        emailDialog->accept();
    });

    connect(cancelBtn, &QPushButton::clicked, emailDialog, &QDialog::reject);

    emailDialog->exec();
    delete emailDialog;
}

void CommandeWidget::onActionExportPDF()
{
    QDialog *exportDialog = new QDialog(this);
    exportDialog->setWindowTitle("Exporter en PDF");
    exportDialog->setMinimumSize(400, 300);

    QVBoxLayout *layout = new QVBoxLayout(exportDialog);

    QLabel *label = new QLabel("Options d'export PDF", exportDialog);

    QGroupBox *optionsGroup = new QGroupBox("Options", exportDialog);
    QVBoxLayout *optionsLayout = new QVBoxLayout(optionsGroup);

    QCheckBox *includeDetails = new QCheckBox("Inclure les d√©tails", optionsGroup);
    QCheckBox *includeStats = new QCheckBox("Inclure les statistiques", optionsGroup);
    QCheckBox *includeCalendar = new QCheckBox("Inclure le calendrier", optionsGroup);

    includeDetails->setChecked(true);
    includeStats->setChecked(true);

    optionsLayout->addWidget(includeDetails);
    optionsLayout->addWidget(includeStats);
    optionsLayout->addWidget(includeCalendar);

    QPushButton *exportBtn = new QPushButton("Exporter", exportDialog);
    QPushButton *cancelBtn = new QPushButton("Annuler", exportDialog);

    QHBoxLayout *buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(exportBtn);
    buttonLayout->addWidget(cancelBtn);

    layout->addWidget(label);
    layout->addWidget(optionsGroup);
    layout->addLayout(buttonLayout);

    connect(exportBtn, &QPushButton::clicked, exportDialog, [exportDialog]() {
        QMessageBox::information(exportDialog, "Export r√©ussi",
                                 "Le PDF a √©t√© export√© avec succ√®s.\nFichier: commandes_export.pdf");
        exportDialog->accept();
    });

    connect(cancelBtn, &QPushButton::clicked, exportDialog, &QDialog::reject);

    exportDialog->exec();
    delete exportDialog;
}

void CommandeWidget::onActionRefresh()
{
    // Simulation de chargement de donn√©es
    setupTableData();
    updateStatistics();
}

void CommandeWidget::onActionStatistics()
{
    // Cr√©er une fen√™tre de dialogue pour afficher les statistiques
    QDialog *statsDialog = new QDialog(this);
    statsDialog->setWindowTitle("Statistiques");
    statsDialog->setMinimumSize(700, 600);

    QVBoxLayout *mainLayout = new QVBoxLayout(statsDialog);

    // Label titre
    QLabel *titleLabel = new QLabel("üìä Statistiques des Commandes", statsDialog);
    titleLabel->setAlignment(Qt::AlignCenter);
    QFont titleFont = titleLabel->font();
    titleFont.setPointSize(14);
    titleFont.setBold(true);
    titleLabel->setFont(titleFont);
    mainLayout->addWidget(titleLabel);

    // GroupBox 0: Statistiques G√©n√©rales (depuis la premi√®re page)
    QGroupBox *groupBoxStats = new QGroupBox("Statistiques G√©n√©rales", statsDialog);
    QGridLayout *gridLayoutStats = new QGridLayout(groupBoxStats);
    
    // Calculer les statistiques
    int total = 0;
    int enCours = 0;
    int validee = 0;
    int prete = 0;
    int annulee = 0;
    
    for (int i = 0; i < ui->tableCommandes->rowCount(); ++i) {
        if (!ui->tableCommandes->isRowHidden(i)) {
            total++;
            QString etat = ui->tableCommandes->item(i, 3)->text();
            if (etat == "En cours") enCours++;
            else if (etat == "Valid√©e") validee++;
            else if (etat == "Pr√™te") prete++;
            else if (etat == "Annul√©e") annulee++;
        }
    }
    
    // Total Commandes
    QLabel *labelStatTotal = new QLabel("Total Commandes :", statsDialog);
    QLabel *statTotal = new QLabel(QString::number(total), statsDialog);
    gridLayoutStats->addWidget(labelStatTotal, 0, 0);
    gridLayoutStats->addWidget(statTotal, 0, 1);
    
    // En cours
    QLabel *labelStatEnCours = new QLabel("En cours :", statsDialog);
    QLabel *statEnCours = new QLabel(QString::number(enCours), statsDialog);
    gridLayoutStats->addWidget(labelStatEnCours, 0, 2);
    gridLayoutStats->addWidget(statEnCours, 0, 3);
    
    // Valid√©es
    QLabel *labelStatValidee = new QLabel("Valid√©es :", statsDialog);
    QLabel *statValidee = new QLabel(QString::number(validee), statsDialog);
    gridLayoutStats->addWidget(labelStatValidee, 1, 0);
    gridLayoutStats->addWidget(statValidee, 1, 1);
    
    // Pr√™tes
    QLabel *labelStatPrete = new QLabel("Pr√™tes :", statsDialog);
    QLabel *statPrete = new QLabel(QString::number(prete), statsDialog);
    gridLayoutStats->addWidget(labelStatPrete, 1, 2);
    gridLayoutStats->addWidget(statPrete, 1, 3);
    
    // Annul√©es
    QLabel *labelStatAnnulee = new QLabel("Annul√©es :", statsDialog);
    QLabel *statAnnulee = new QLabel(QString::number(annulee), statsDialog);
    gridLayoutStats->addWidget(labelStatAnnulee, 2, 0);
    gridLayoutStats->addWidget(statAnnulee, 2, 1);
    
    mainLayout->addWidget(groupBoxStats);

    // GroupBox 1: Produits par Type/Stock
    QGroupBox *groupBoxStatsByType = new QGroupBox("Produits par Type/Stock", statsDialog);
    QVBoxLayout *statsTypesLayout = new QVBoxLayout(groupBoxStatsByType);
    QTextEdit *statsTypesEdit = new QTextEdit(statsDialog);
    statsTypesEdit->setReadOnly(true);
    
    QString statsTypesText = "=== ANALYSE PAR TYPE DE PRODUIT ===\n\n"
                             "‚Ä¢ Huiles: 45 unit√©s (Stock: Normal)\n"
                             "‚Ä¢ Machines: 12 unit√©s (Stock: Bas)\n"
                             "‚Ä¢ Emballages: 120 unit√©s (Stock: √âlev√©)\n"
                             "‚Ä¢ Pi√®ces d√©tach√©es: 67 unit√©s (Stock: Normal)\n\n"
                             "=== ALERTES STOCK ===\n"
                             "‚ö† 2 produits en niveau critique\n"
                             "‚ö† 5 produits √† r√©approvisionner";
    statsTypesEdit->setPlainText(statsTypesText);
    statsTypesLayout->addWidget(statsTypesEdit);
    mainLayout->addWidget(groupBoxStatsByType);

    // GroupBox 2: D√©penses par Mois / Ruptures
    QGroupBox *groupBoxStatsDepenses = new QGroupBox("D√©penses par Mois / Ruptures", statsDialog);
    QVBoxLayout *statsDepensesLayout = new QVBoxLayout(groupBoxStatsDepenses);
    QTextEdit *statsDepensesEdit = new QTextEdit(statsDialog);
    statsDepensesEdit->setReadOnly(true);
    
    QString statsDepensesText = "=== D√âPENSES MENSUELLES 2024 ===\n\n"
                                "‚Ä¢ Janvier: 12,500‚Ç¨\n"
                                "‚Ä¢ F√©vrier: 14,200‚Ç¨\n"
                                "‚Ä¢ Mars: 11,800‚Ç¨\n"
                                "‚Ä¢ Avril: 13,100‚Ç¨\n\n"
                                "=== TENDANCE ===\n"
                                "üìà Augmentation de 8% vs 2023\n\n"
                                "=== RUPTURES ===\n"
                                "‚Ä¢ Produit A: 2 occurrences\n"
                                "‚Ä¢ Produit B: 1 occurrence\n"
                                "‚Ä¢ Produit C: 0 occurrence";
    statsDepensesEdit->setPlainText(statsDepensesText);
    statsDepensesLayout->addWidget(statsDepensesEdit);
    mainLayout->addWidget(groupBoxStatsDepenses);

    // Bouton Fermer
    QPushButton *closeBtn = new QPushButton("Fermer", statsDialog);
    connect(closeBtn, &QPushButton::clicked, statsDialog, &QDialog::accept);
    mainLayout->addWidget(closeBtn);

    statsDialog->exec();
    delete statsDialog;
}

// ==================== QUICK ACTION BUTTONS ====================

void CommandeWidget::onBtnNewCommandeClicked()
{
    onActionNew();
}

void CommandeWidget::onBtnEditCommandeClicked()
{
    onActionEdit();
}

void CommandeWidget::onActionCalendar()
{
    // Cr√©er une fen√™tre de dialogue pour afficher le calendrier
    QDialog *calendarDialog = new QDialog(this);
    calendarDialog->setWindowTitle("Calendrier des Commandes");
    calendarDialog->setMinimumSize(500, 500);

    QVBoxLayout *mainLayout = new QVBoxLayout(calendarDialog);

    // Label titre
    QLabel *titleLabel = new QLabel("üìÖ Calendrier des Commandes", calendarDialog);
    titleLabel->setAlignment(Qt::AlignCenter);
    QFont titleFont = titleLabel->font();
    titleFont.setPointSize(14);
    titleFont.setBold(true);
    titleLabel->setFont(titleFont);
    mainLayout->addWidget(titleLabel);

    // Calendrier
    QCalendarWidget *calendar = new QCalendarWidget(calendarDialog);
    
    // Marquer des dates importantes
    QTextCharFormat importantFormat;
    importantFormat.setBackground(QColor("#BDD2B6"));
    importantFormat.setFontWeight(QFont::Bold);
    
    QDate today = QDate::currentDate();
    calendar->setDateTextFormat(today.addDays(3), importantFormat);
    calendar->setDateTextFormat(today.addDays(10), importantFormat);
    calendar->setDateTextFormat(today.addDays(15), importantFormat);
    
    mainLayout->addWidget(calendar);

    // Bouton Fermer
    QPushButton *closeBtn = new QPushButton("Fermer", calendarDialog);
    connect(closeBtn, &QPushButton::clicked, calendarDialog, &QDialog::accept);
    mainLayout->addWidget(closeBtn);

    calendarDialog->exec();
    delete calendarDialog;
}

void CommandeWidget::onBtnDeleteCommandeClicked()
{
    onActionDelete();
}

void CommandeWidget::onBtnEnvoyerEmailClicked()
{
    onActionSendEmail();
}

// ==================== SEARCH AND FILTERS ====================

void CommandeWidget::onSearchTextChanged(const QString &text)
{
    applyFilters();
}

void CommandeWidget::onFilterChanged()
{
    applyFilters();
}

void CommandeWidget::applyFilters()
{
    QString searchText = ui->lineEditSearch->text().toLower();
    QString selectedFournisseur = ui->comboFournisseur->currentText();
    QString selectedStatus = ui->comboStatus->currentText();
    QDate dateFrom = ui->dateFrom->date();
    QDate dateTo = ui->dateTo->date();

    for (int row = 0; row < ui->tableCommandes->rowCount(); ++row) {
        bool shouldShow = true;

        // Filtrer par texte de recherche
        if (!searchText.isEmpty()) {
            bool found = false;
            for (int col = 0; col < ui->tableCommandes->columnCount(); ++col) {
                QTableWidgetItem *item = ui->tableCommandes->item(row, col);
                if (item && item->text().toLower().contains(searchText)) {
                    found = true;
                    break;
                }
            }
            shouldShow = shouldShow && found;
        }

        // Filtrer par fournisseur
        if (selectedFournisseur != "Toutes") {
            QTableWidgetItem *fournisseurItem = ui->tableCommandes->item(row, 2);
            if (fournisseurItem && fournisseurItem->text() != selectedFournisseur) {
                shouldShow = false;
            }
        }

        // Filtrer par statut
        if (selectedStatus != "Tous") {
            QTableWidgetItem *statusItem = ui->tableCommandes->item(row, 3);
            if (statusItem && statusItem->text() != selectedStatus) {
                shouldShow = false;
            }
        }

        // Filtrer par date
        QTableWidgetItem *dateItem = ui->tableCommandes->item(row, 4);
        if (dateItem && !dateItem->text().isEmpty()) {
            QDate orderDate = QDate::fromString(dateItem->text(), "yyyy-MM-dd");
            if (orderDate < dateFrom || orderDate > dateTo) {
                shouldShow = false;
            }
        }

        // Afficher ou masquer la ligne
        ui->tableCommandes->setRowHidden(row, !shouldShow);
    }

    updateStatistics();
}

// ==================== HELPER METHODS ====================

void CommandeWidget::updateQuickInfo()
{
    int currentRow = ui->tableCommandes->currentRow();
    if (currentRow >= 0 && !ui->tableCommandes->isRowHidden(currentRow)) {
        ui->quickEtat->setText(ui->tableCommandes->item(currentRow, 3)->text());
        ui->quickFournisseur->setText(ui->tableCommandes->item(currentRow, 2)->text());
        ui->quickDateArrivee->setText(ui->tableCommandes->item(currentRow, 4)->text());
        ui->quickDateSortie->setText(ui->tableCommandes->item(currentRow, 5)->text());
        ui->quickMontant->setText(ui->tableCommandes->item(currentRow, 6)->text());
    } else {
        ui->quickEtat->setText("-");
        ui->quickFournisseur->setText("-");
        ui->quickDateArrivee->setText("-");
        ui->quickDateSortie->setText("-");
        ui->quickMontant->setText("-");
    }
}

void CommandeWidget::updateStatistics()
{
    // Les statistiques sont maintenant affich√©es dans la fen√™tre "Statistiques" au lieu de la premi√®re page
    // Cette fonction reste ici pour les futurs traitements si n√©cessaire
}

void CommandeWidget::showCrudDialog(const QString &title)
{
    QDialog *dialog = new QDialog(this);
    dialog->setWindowTitle(title);
    dialog->setMinimumSize(500, 400);

    QVBoxLayout *mainLayout = new QVBoxLayout(dialog);

    QLabel *titleLabel = new QLabel(title, dialog);
    titleLabel->setAlignment(Qt::AlignCenter);
    QFont titleFont = titleLabel->font();
    titleFont.setPointSize(14);
    titleFont.setBold(true);
    titleLabel->setFont(titleFont);

    QFormLayout *formLayout = new QFormLayout();

    // Champs g√©n√©riques selon le type
    if (title.contains("Commande")) {
        formLayout->addRow("ID Commande:", new QLineEdit(dialog));
        formLayout->addRow("Libell√©:", new QLineEdit(dialog));

        QComboBox *fournisseurCombo = new QComboBox(dialog);
        fournisseurCombo->addItems({"Fournisseur A", "Fournisseur B", "Fournisseur C", "Fournisseur D"});
        formLayout->addRow("Fournisseur:", fournisseurCombo);

        QComboBox *statusCombo = new QComboBox(dialog);
        statusCombo->addItems({"En cours", "Valid√©e", "Pr√™te", "Annul√©e"});
        formLayout->addRow("√âtat:", statusCombo);

        formLayout->addRow("Date Arriv√©e:", new QDateEdit(QDate::currentDate(), dialog));
        formLayout->addRow("Date Sortie:", new QDateEdit(QDate::currentDate().addDays(7), dialog));

        QDoubleSpinBox *montantSpin = new QDoubleSpinBox(dialog);
        montantSpin->setRange(0, 100000);
        montantSpin->setPrefix("‚Ç¨ ");
        formLayout->addRow("Montant:", montantSpin);
    } else if (title.contains("Employ√©")) {
        formLayout->addRow("Nom:", new QLineEdit(dialog));
        formLayout->addRow("Pr√©nom:", new QLineEdit(dialog));
        formLayout->addRow("Email:", new QLineEdit(dialog));
        formLayout->addRow("T√©l√©phone:", new QLineEdit(dialog));

        QComboBox *posteCombo = new QComboBox(dialog);
        posteCombo->addItems({"Gestionnaire", "Technicien", "Commercial", "Administratif"});
        formLayout->addRow("Poste:", posteCombo);
    } else {
        formLayout->addRow("Nom:", new QLineEdit(dialog));
        QTextEdit *descEdit = new QTextEdit(dialog);
        descEdit->setMaximumHeight(100);
        formLayout->addRow("Description:", descEdit);
    }

    QPushButton *saveBtn = new QPushButton("Enregistrer", dialog);
    QPushButton *cancelBtn = new QPushButton("Annuler", dialog);

    QHBoxLayout *buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(saveBtn);
    buttonLayout->addWidget(cancelBtn);

    mainLayout->addWidget(titleLabel);
    mainLayout->addLayout(formLayout);
    mainLayout->addStretch();
    mainLayout->addLayout(buttonLayout);

    connect(saveBtn, &QPushButton::clicked, dialog, [dialog, this]() {
        QMessageBox::information(dialog, "Succ√®s", "Enregistrement effectu√© avec succ√®s.");
        dialog->accept();
        onActionRefresh(); // Rafra√Æchir les donn√©es
    });

    connect(cancelBtn, &QPushButton::clicked, dialog, &QDialog::reject);

    dialog->exec();
    delete dialog;
}

void CommandeWidget::showAdvancedFeatureDialog(const QString &title, const QString &description)
{
    QDialog *dialog = new QDialog(this);
    dialog->setWindowTitle(title);
    dialog->setMinimumSize(700, 500);

    QVBoxLayout *mainLayout = new QVBoxLayout(dialog);

    // En-t√™te
    QLabel *titleLabel = new QLabel(title, dialog);
    titleLabel->setAlignment(Qt::AlignCenter);
    QFont titleFont = titleLabel->font();
    titleFont.setPointSize(16);
    titleFont.setBold(true);
    titleLabel->setFont(titleFont);
    titleLabel->setStyleSheet("color: #798777; padding: 10px;");

    // Description
    QLabel *descLabel = new QLabel(description, dialog);
    descLabel->setAlignment(Qt::AlignLeft);
    descLabel->setWordWrap(true);
    descLabel->setStyleSheet("background-color: #F8EDE3; padding: 15px; border-radius: 5px;");

    // Onglets pour diff√©rentes vues
    QTabWidget *tabWidget = new QTabWidget(dialog);

    // Onglet 1: Vue d√©taill√©e
    QWidget *tab1 = new QWidget();
    QVBoxLayout *tab1Layout = new QVBoxLayout(tab1);

    QTableWidget *detailTable = new QTableWidget(10, 5, tab1);
    QStringList headers = {"M√©trique", "Valeur", "√âvolution", "Seuil", "Statut"};
    detailTable->setHorizontalHeaderLabels(headers);
    tab1Layout->addWidget(detailTable);
    tabWidget->addTab(tab1, "Vue d√©taill√©e");

    // Onglet 2: Graphiques
    QWidget *tab2 = new QWidget();
    QVBoxLayout *tab2Layout = new QVBoxLayout(tab2);

    QLabel *graphLabel = new QLabel("üìà Visualisation graphique des donn√©es\n\n"
                                    "‚Ä¢ Graphique 1: √âvolution des stocks\n"
                                    "‚Ä¢ Graphique 2: D√©penses mensuelles\n"
                                    "‚Ä¢ Graphique 3: Tendance des commandes", tab2);
    graphLabel->setAlignment(Qt::AlignCenter);
    graphLabel->setMinimumHeight(300);
    graphLabel->setStyleSheet("background-color: #BDD2B6; border-radius: 5px; padding: 20px;");
    tab2Layout->addWidget(graphLabel);
    tabWidget->addTab(tab2, "Graphiques");

    // Onglet 3: Export/Rapports
    QWidget *tab3 = new QWidget();
    QVBoxLayout *tab3Layout = new QVBoxLayout(tab3);

    QGroupBox *exportGroup = new QGroupBox("Options d'export", tab3);
    QVBoxLayout *exportLayout = new QVBoxLayout(exportGroup);

    exportLayout->addWidget(new QCheckBox("Exporter en PDF"));
    exportLayout->addWidget(new QCheckBox("Exporter en Excel"));
    exportLayout->addWidget(new QCheckBox("Exporter en CSV"));
    exportLayout->addWidget(new QCheckBox("G√©n√©rer un rapport automatique"));

    tab3Layout->addWidget(exportGroup);
    tabWidget->addTab(tab3, "Export/Rapports");

    // Boutons
    QPushButton *closeBtn = new QPushButton("Fermer", dialog);
    QPushButton *generateBtn = new QPushButton("G√©n√©rer Rapport", dialog);

    QHBoxLayout *buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(generateBtn);
    buttonLayout->addStretch();
    buttonLayout->addWidget(closeBtn);

    mainLayout->addWidget(titleLabel);
    mainLayout->addWidget(descLabel);
    mainLayout->addWidget(tabWidget);
    mainLayout->addLayout(buttonLayout);

    connect(generateBtn, &QPushButton::clicked, dialog, [dialog]() {
        QMessageBox::information(dialog, "Rapport g√©n√©r√©",
                                 "Le rapport avanc√© a √©t√© g√©n√©r√© avec succ√®s.\n"
                                 "Consultez le dossier 'Rapports' pour plus de d√©tails.");
    });

    connect(closeBtn, &QPushButton::clicked, dialog, &QDialog::accept);

    dialog->exec();
    delete dialog;
}
